AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/CrashLogging.kt

```diff
@@ -39,41 +47,25 @@ object CrashLogging {
                 setDebug(dataProvider.enableCrashLoggingLogs)
                 setTag("locale", dataProvider.locale?.language ?: "unknown")
                 beforeSend = SentryOptions.BeforeSendCallback { event, _ ->
-                    return@BeforeSendCallback if (dataProvider.userHasOptedOut) {
+                    return@BeforeSendCallback if (userOptOut) {
                         null
                     } else {
-                        event
+                        event.apply {
+                            user = dataProvider.userProvider()?.toSentryUser()
+                        }
```

Comment:
This is inspired by the current implementation on `WPAndroid`. It takes the responsibility of the client to update the state of current user data. Instead, before every event, a correct user is applied.

https://github.com/wordpress-mobile/WordPress-Android/blob/a9eeed79f6d85fd3a3a55157ff9b2932248c00cc/WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt#L36-L42